addon/scheduler.ts

```diff
@@ -13,32 +14,80 @@ interface Deferred {
   reject: Function;
 }
 
+interface SchedulerOptions {
+  markName: string;
+  emitMark: boolean;
+}
+
 interface Capabilities {
   requestAnimationFrameEnabled: boolean;
   requestIdleCallbackEnabled: boolean;
+  performanceObserverEnabled: boolean;
 }
 
 const APP_SCHEDULER_LABEL: string = 'ember-app-scheduler';
 const APP_SCHEDULER_HAS_SETUP: string = '__APP_SCHEDULER_HAS_SETUP__';
+let PERFORMANCE_OBSERVER_SETUP: boolean = false;
 
 let _whenRouteDidChange: Deferred;
 let _whenRoutePainted: Promise<any>;
 let _whenRoutePaintedScheduleFn: Function;
 let _whenRouteIdle: Promise<any>;
 let _whenRouteIdleScheduleFn: Function;
-let _activeScheduledTasks: number = 0;
 const CAPABILITIES: Capabilities = {
   requestAnimationFrameEnabled: typeof requestAnimationFrame === 'function',
   requestIdleCallbackEnabled: typeof requestIdleCallback === 'function',
+  performanceObserverEnabled: typeof PerformanceObserver === 'function',
+};
+export const DEFAULT_MARK_NAME: string = 'routeIdle';
```

Comment:
This is not used outside of `DEFAULT_OPTIONS`, do we really need a separate variable for it just for "dry" tests?

Reply:
Yerp, this is likely on the 'chapped' side. Will remove.